include/swift/AST/DiagnosticsParse.def

```diff
@@ -724,6 +724,8 @@ ERROR(rethrows_in_wrong_position,none,
       "'rethrows' may only occur before '->'", ())
 ERROR(throw_in_function_type,none,
       "expected throwing specifier; did you mean 'throws'?", ())
+NOTE(throw_in_function_type_and_wrong_position_note,none,
+      "expected throwing specifier; did you mean 'throws'?", ())
```

Comment:
Let’s not surface the word “specifier” to users here.  You can remove the first clause before the semicolon and just use the dependent clause for the fixit.

Reply:
I think we can be more specific.  The dependent clause is good, but the clause before the semicolon doesn’t actually reflect the mistake the user made.

Reply:
What do you prefer instead then?

The user has made two mistakes (wrong keyword and in wrong position) and I think both should be surfaced.

Right now 1 is displayed in the error and the other in the note. 
In my original commit the error displayed both at the same time.

Reply:
I would expect one error message only to explain the problem. Something like, "unexpected 'throw' in function type; did you mean 'throws', which may occur only before '->'?"

Agree that "specifier" isn't a good user-facing word and the existing diagnostic could use improvement too.

Reply:
> I would expect one error message only to explain the problem.

That was my initial expectation also :) In case we want just one error I would rollback to 592d660 and apply the copy changes, which look like an improvement anyway 👍 

If somebody can confirm which behaviour we prefer (1 errors vs. error + note) I can continue with this.